src/libraries/System.Net.Security/src/System/Net/Security/Pal.Android/SafeDeleteSslContext.cs

```diff
@@ -45,6 +51,7 @@ public SafeDeleteSslContext(SslAuthenticationOptions authOptions)
             try
             {
                 _sslContext = CreateSslContext(SslStreamProxy, authOptions);
+                _gcHandle = GCHandle.Alloc(this);
```

Comment:
This will root the instance until `Dispose` is explicitly called. In case the user doesn't call it (not unheard of), GC will not release it ever, regardless of finalizer existence or not.

We had a similar problem in Quic with MsQuic and we're now only rooting the object for the limited time when the control is in native code.

Reply:
Thanks! Since it’s used in callbacks, we need to keep a strong reference to it. I introduced a static dictionary to hold the reference, which is removed during disposal.

Reply:
Update: No GCHandle reference from managed side is needed.